--- a/engines/sherlock/tattoo/tattoo_people.cpp
+++ b/engines/sherlock/tattoo/tattoo_people.cpp
@@ -479,22 +479,22 @@ void TattooPerson::setWalking() {
 		} else {
 			// Major movement is vertical, so set the sequence for up and down,
 			// and set the delta Y depending on the direction
 			if (_walkDest.y < (_position.y / FIXED_INT_MULTIPLIER)) {
 				_sequenceNumber = WALK_UP;
 				_delta.y = speed.y * -(FIXED_INT_MULTIPLIER / 10);
 			} else {
 				speed.y = diagSpeed.y;
 				_sequenceNumber = WALK_DOWN;
 				_delta.y = speed.y * (FIXED_INT_MULTIPLIER / 10);
 			}
 
 			// Set the delta x
 			if (delta.y * 10 / speed.y)
 				_delta.x = (delta.x * FIXED_INT_MULTIPLIER) / (delta.y * 10 / speed.y);
 			else
 				_delta.x = (delta.x * FIXED_INT_MULTIPLIER) / delta.y;
 
-			if (_walkDest.x < _position.y / FIXED_INT_MULTIPLIER)
+			if (_walkDest.x < _position.x / FIXED_INT_MULTIPLIER)
 				_delta.x = -_delta.x;
 
 			// Set how many times we should add the delta's to the players position
